hudi-client/hudi-client-common/src/main/java/org/apache/hudi/index/bucket/ConsistentBucketIndexUtils.java

```diff
@@ -108,11 +108,11 @@ public static Option<HoodieConsistentHashingMetadata> loadMetadata(HoodieTable t
     try {
       Predicate<StoragePathInfo> hashingMetaCommitFilePredicate = pathInfo -> {
         String filename = pathInfo.getPath().getName();
-        return filename.contains(HoodieConsistentHashingMetadata.HASHING_METADATA_COMMIT_FILE_SUFFIX);
+        return filename.endsWith(HoodieConsistentHashingMetadata.HASHING_METADATA_COMMIT_FILE_SUFFIX);
```

Comment:
using `HoodieStorage::createImmutableFileInPath` will create a temp file with UUID-endifx, so we should ignore these temp files.